base/reduce.jl

```diff
@@ -861,3 +861,19 @@ function count(pred, a::AbstractArray)
     return n
 end
 count(itr) = count(identity, itr)
+
+function count(::typeof(identity), x::Array{Bool})
+    n = 0
+    chunks = length(x) ÷ sizeof(UInt)
+    mask = 0x0101010101010101 % UInt
+    GC.@preserve x begin
+        ptr = Ptr{UInt}(pointer(x))
+        for i in 1:chunks
```

Comment:
Does `@simd` help?

Reply:
It shouldn't for integers, should it?

Reply:
> It shouldn't for integers, should it?

I had the same thought, but it actually did help in my `count4` implementation above.  I think that's due to the different lowering of the iteration space for a step-range, though.

Given the timings and LLVM code for this implementation, I'd be very surprised if it did anything further.